src/terrain/draw_terrain_raster.js

```diff
@@ -161,6 +162,30 @@ function drawTerrainForGlobe(painter: Painter, terrain: Terrain, sourceCache: So
     const mercatorCenter = [mercatorXfromLng(tr.center.lng), mercatorYfromLat(tr.center.lat)];
     const batches = showWireframe ? [false, true] : [false];
     const sharedBuffers = painter.globeSharedBuffers;
+    const isAntialias = !painter.style.map._antialias;
+
+    // Render the gradient atmosphere by casting rays from screen pixels and determining their
+    // closest distance to the globe. This is done in view space where camera is located in the origo
+    // facing -z direction.
+    const viewMatrix = tr._camera.getWorldToCamera(tr.worldSize, 1.0);
+    const viewToProj = tr._camera.getCameraToClipPerspective(tr._fov, tr.width / tr.height, tr._nearZ, tr._farZ);
+    const projToView = mat4.invert([], viewToProj);
+
+    const project = (point: Vec3, m: Mat4): [number, number, number] => {
+        vec3.transformMat4(point, point, m);
+        return [point[0], point[1], point[2]];
+    };
+
+    // Compute direction vectors to each corner point of the view frustum
+    const frustumTl = project([-1, 1, 1], projToView);
+    const frustumTr = project([1, 1, 1], projToView);
+    const frustumBr = project([1, -1, 1], projToView);
+    const frustumBl = project([-1, -1, 1], projToView);
+
+    const center = getColumn(tr.globeMatrix, 3);
+    const globeCenterInViewSpace = project([center[0], center[1], center[2]], viewMatrix);
+    const globeRadius = tr.worldSize / 2.0 / Math.PI - 1.0;
+    const viewport = [tr.width, tr.height];
```

Comment:
I've recently updated this code so that these variable are calculated only once per frame (Refer https://github.com/mapbox/mapbox-gl-js/commit/6971327e188b9aa045622925a59800aa8ee940ac), so you can now retrieve them from the transform for simplicity:

```
tr.frustumCorners.TL,
tr.frustumCorners.TR,
tr.frustumCorners.BR,
tr.frustumCorners.BL,
tr.globeCenterInViewSpace,
tr.globeRadius,
[
    tr.width * browser.devicePixelRatio,
    tr.height * browser.devicePixelRatio
]
```

Reply:
done!